pulumi/main.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
+
+	"github.com/pulumi/pulumi-random/sdk/v4/go/random"
+
+	"github.com/threefoldtech/pulumi-threefold/sdk/go/threefold"
+	"github.com/threefoldtech/pulumi-threefold/sdk/go/threefold/provider"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+
+		grid_provider, err := threefold.NewProvider(ctx, "grid provider", &threefold.ProviderArgs{
+			Mnemonic: config.RequireSecret(ctx, "tfgrid_devnet_mnemonic"),
+			Network:  pulumi.String("dev"),
+		})
+		if err != nil {
+			return err
+		}
+
+		const node int = 195
+
+		mycelium_seed, err := random.NewRandomBytes(ctx, "mycelium_seed", &random.RandomBytesArgs{
+			Length: pulumi.Int(32),
+		})
+		if err != nil {
+			return err
+		}
+
+		mycelium_ip_seed, err := random.NewRandomBytes(ctx, "mycelium_ip_seed", &random.RandomBytesArgs{
+			Length: pulumi.Int(6),
+		})
+		if err != nil {
+			return err
+		}
+
+		grid_network, err := provider.NewNetwork(ctx, "network0", &provider.NetworkArgs{
+			Description: pulumi.String("example grid network 0"),
+			Ip_range:    pulumi.String("10.1.0.0/16"),
```

Comment:
doesn't look like it in the [data structure docs](https://pkg.go.dev/github.com/threefoldtech/tfgrid-sdk-go/grid-client/workloads#ZNet).

Reply:
the wireguard access is authenticated by design. the only publicly reachable resource in our case is the mycelium IPv6 address. if we wanted to constraint access to that we'd look for options on mycelium itself, and then there are the firewall settings of the deployed VM.

the flist is created from [the tfgrid-base config](https://github.com/holochain/holochain-infra/blob/develop/modules/flake-parts/nixosConfigurations.tfgrid-base/configuration.nix), which is the default and looks like this:

```json
$ nix eval --json .\#nixosConfigurations.tfgrid-base.config.networking.firewall | nix run nixpkgs#jq
{
  "allInterfaces": {
    "default": {
      "allowedTCPPortRanges": [],
      "allowedTCPPorts": [
        22
      ],
      "allowedUDPPortRanges": [],
      "allowedUDPPorts": []
    }
  },
  "allowPing": true,
  "allowedTCPPortRanges": [],
  "allowedTCPPorts": [
    22
  ],
  "allowedUDPPortRanges": [],
  "allowedUDPPorts": [],
  "autoLoadConntrackHelpers": false,
  "checkReversePath": true,
  "connectionTrackingModules": [],
  "enable": true,
  "extraCommands": "# Helper command to manipulate both the IPv4 and IPv6 tables.\nip46tables() {\n  iptables -w \"$@\"\n  ip6tables -w \"$@\"\n\n}\n\nip46tables -w -t nat -D PREROUTING -j nixos-nat-pre 2>/dev/null|| true\nip46tables -w -t nat -F nixos-nat-pre 2>/dev/null || true\nip46tables -w -t nat -X nixos-nat-pre 2>/dev/null || true\nip46tables -w -t nat -D POSTROUTING -j nixos-nat-post 2>/dev/null || true\nip46tables -w -t nat -F nixos-nat-post 2>/dev/null || true\nip46tables -w -t nat -X nixos-nat-post 2>/dev/null || true\nip46tables -w -t nat -D OUTPUT -j nixos-nat-out 2>/dev/null || true\nip46tables -w -t nat -F nixos-nat-out 2>/dev/null || true\nip46tables -w -t nat -X nixos-nat-out 2>/dev/null || true\n\n\n",
  "extraForwardRules": "",
  "extraInputRules": "",
  "extraPackages": [],
  "extraStopCommands": "",
  "filterForward": false,
  "interfaces": {},
  "logRefusedConnections": false,
  "logRefusedPackets": false,
  "logRefusedUnicastsOnly": true,
  "logReversePathDrops": false,
  "package": "/nix/store/ncgdax7xgk3lj499jkb52zksf98c83bq-iptables-1.8.10",
  "pingLimit": null,
  "rejectPackets": false,
  "trustedInterfaces": [
    "lo"
  ]
}

as you can see here, it allows ingress on port 22, that's from any IP. i think that's secure enough because SSH authentication is sufficiently secure by itself, see the [holo-users module](https://github.com/holochain/holochain-infra/blob/develop/modules/flake-parts/nixosModules.holo-users.nix) for how it works in this repo.